### PR TITLE
[WIP] docs: improve and simplify documentation

### LINKED ISSUE #16350: Contribution guidelines and workflows need clarifications.
#### Issue Description
I have noticed over the past week and a half that it is difficult to find issues to work on without stepping on toes or waiting to hear from someone several hours later if it's okay to work on it. In many instances, I've ended up playing with something that was marked as Help Wanted and unassigned and in the Todo bucket, only to realize that someone else was working on it!

Work-items should reflect current and accurate status. I would like to see the following:

1. People working on issues need to be assigned to it, or at least a note stating that a specific person is investigating needs to be posted.

2. The issue must be moved to in-progress (or in an investigating bucket)

3. The issue needs to be tagged appropriately

I spent a week trying to get started because I couldn't find an available issue to work on and almost gave up.

### LINKED ISSUE #16373: Update docs about generating ID for New Challenges
We need to update the [Challenge Style Guide](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/seed/challenge-style-guide.md) and refer to it prominently from the [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors).

The update should be along the lines of:

----
To generate a new one in a shell (assuming MongoDB is running separately):

1. Run `mongo` command
2. Run `ObjectId()` command

Output:

```sh
$ mongo
MongoDB shell version v3.6.1
connecting to: mongodb://127.0.0.1:27017
MongoDB server version: 3.4.10
...
$ ObjectId()
ObjectId("5a474d78df58bafeb3535d34")
```

The result is a new id, for example `"5a474d78df58bafeb3535d34"` above.

----
Refers: https://github.com/freeCodeCamp/freeCodeCamp/pull/16369#issuecomment-354534248

### PR DESCRIPTION
**Pull request by raisedadead**
Closes #16373
Closes #16350

### CODE REVIEW

**Comment by vkWeb** [comment]
@raisedadead

**I think we should improve our doc in these areas:**
- We should also tell our contributors to open `mongo` in another terminal after running `mongod`. I've faced a lot of troubles setting up the development environment with MongoDB(Windows 32 Bit). I'm not sure about this but running `mongo` after `mongod` worked for me. 

- We should inform our campers to include `C:\Program Files\MongoDB\Server\3.2\bin` in their environmental variables so that they don't get an error while running `mongo --version` as we're telling them to run in the beginning to check the version. 

I've no experience with mongo or MongoDB. This PR was open so I thought I should nudge about this.

**Comment by BerkeleyTrue** [comment]
> We should also tell our contributors to open...

> We should inform our campers to include....

@vkWeb Thanks for the input but these are specific to running Mongo and including them in the docs would make them extremely brittle. Including links to Mongo's official documentation is sufficient enough.

**Comment by vkWeb** [comment]
@BerkeleyTrue Okay 👍.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,495 +1 @@
...
-
--   [How to log in to your local freeCodeCamp site - using
-    GitHub](http://forum.freecodecamp.org/t/how-to-log-in-to-your-local-instance-of-free-code-camp/19552)
-
--   [Writing great git commit
-    messages](http://forum.freecodecamp.org/t/writing-good-git-commit-messages/13210)
-
--   [Contributor Chat Support - For the freeCodeCamp repositories, and running a local
-    instance](https://gitter.im/FreeCodeCamp/Contributors)
+> Note: This guide is now [moved to the docs directory](/docs/).
```
Comment (systimotic) [inline CONTRIBUTING.md]: I think these are some really good improvements 👍 My only concern is that GitHub automatically refers to the CONTRIBUTING.md when creating a new PR or issue. If we link them to the folder, they might not now which file they're looking for.

Comment (raisedadead) [reply CONTRIBUTING.md]: Yes, you are correct. That is the prime reason I am keeping this file and linking it to docs, I think there is worth in keeping all documentation neatly organised in docs folder.

Sure, many contributors will land on this file, from GitHub interface in create new issue page mostly,
but we can link to the correct files, from other places, for example the contribute.freecodecamp.org and the readme for example, which gets the most traffic.

**Comment by raisedadead** [comment]
I am going to close this, and make a fresh PR. @alexch has taken up notable work. And this can be deferred to until later.

**Pull request closed without merging**